cloudinit/util.py

```diff
@@ -299,6 +299,26 @@ def rand_str(strlen=32, select_from=None):
     return "".join([r.choice(select_from) for _x in range(strlen)])
 
 
+def get_str_class_num(str):
+    str_class_num = 0
+    if any(c.islower() for c in str):
+        str_class_num += 1
+    if any(c.isupper() for c in str):
+        str_class_num += 1
+    if any(c.isdigit() for c in str):
+        str_class_num += 1
+    if any(c in string.punctuation for c in str):
+        str_class_num += 1
+    return str_class_num
+
+
+def rand_str_minclass(strlen=32, select_from=None, minclass=3):
+    while True:
```

Comment:
I'd prefer a solution that looks something [this this](https://stackoverflow.com/a/33775065) rather than looping an indeterminate number of times.